### PR TITLE
Move to latest tar and fix some bugs / reduce patches

### PR DESCRIPTION
**Pull request by MikeFultonDev**
Tests that _pass_ in the 9.1 run but _fail_ in the 9.4 run are:

```
tests/misc/basename.pl						tests/misc/basename.pl
tests/misc/dirname.pl						tests/misc/dirname.pl
tests/misc/realpath.sh						tests/misc/realpath.sh
tests/misc/tee.sh						tests/misc/tee.sh
```

Of these, only _realpath_ is installed. The failure with realpath has to do with a '//' path which we process in a special manner on z/OS and is not a valid POSIX file name, so risk seems low.

### CODE REVIEW

File: stable-patches/lib/posix_memalign.c.patch [hunk 1]
```diff
@@ -0,0 +1,22 @@
+diff --git a/lib/posix_memalign.c b/lib/posix_memalign.c
+index b490665..6c10626 100644
+--- a/lib/posix_memalign.c
++++ b/lib/posix_memalign.c
+@@ -25,6 +25,11 @@ int
+ posix_memalign (void **memptr, size_t alignment, size_t size)
+ #undef posix_memalign
+ {
++/* FIXME: MVS does not have posix_memalign */
```
Comment (IgorTodorovskiIBM) [inline stable-patches/lib/posix_memalign.c.patch:9]: This is a good candidate for zoslib, seen in a few places, including nginx most recently

File: stable-patches/lib/getprogname.c.patch [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+                   if (s)
+                     {
+-#  if defined __XPLINK__ && __CHARSET_LIB == 1
++#if (__CHARSET_LIB == 1)
+                       /* The compiler option -qascii is in use.
+                          https://makingdeveloperslivesbetter.wordpress.com/2022/01/07/is-z-os-ascii-or-ebcdic-yes/
+                          https://www.ibm.com/docs/en/zos/2.5.0?topic=features-macros-related-compiler-option-settings
+                          So, convert the result from EBCDIC to ASCII.
+                          https://www.ibm.com/docs/en/zos/2.5.0?topic=functions-e2a-s-convert-string-from-ebcdic-ascii */
++#ifdef OLD_RUNTIME
```
Comment (IgorTodorovskiIBM) [inline stable-patches/lib/getprogname.c.patch:16]: minor: Should we guard on __llvm__ or does xlclang experience the same issue?

**Review by IgorTodorovskiIBM — APPROVED** [review]
LGTM!

File: stable-patches/lib/getprogname.c.patch (hunk 2 above)
Comment (MikeFultonDev) [reply stable-patches/lib/getprogname.c.patch:16]: xlclang had the same issue

File: stable-patches/lib/posix_memalign.c.patch (hunk 1 above)
Comment (MikeFultonDev) [reply stable-patches/lib/posix_memalign.c.patch:9]: agree

File: stable-patches/src/copy.c.patch [hunk 3]
```diff
@@ -0,0 +1,35 @@
+diff --git a/src/copy.c b/src/copy.c
+index b9fff03..4683399 100644
+--- a/src/copy.c
++++ b/src/copy.c
+@@ -1216,6 +1216,18 @@ handle_clone_fail (int dst_dirfd, char const *dst_relname,
+   return true;
+ }
+ 
++#ifdef __MVS__
++int __setfdccsid(int fd, int t_ccsid)
```
[the remark this answers is not in the document]
Comment (MikeFultonDev) [reply stable-patches/src/copy.c.patch:10]: nice. so __setfdccsid (or equivalent) is exposed as a function in zoslib?

[the remark this answers is not in the document]
Comment (IgorTodorovskiIBM) [reply stable-patches/src/copy.c.patch:10]: yep in here: https://github.com/ibmruntimes/zoslib/blob/zopen/src/zos-io.cc#L674

**Pull request merged**